### PR TITLE
Adding installHint field to kubeconfigs that have been converted to the exec format

### PR DESCRIPTION
**Pull request by cirvine-MSFT**
InstallHint field tells users how to install the binary in the command field if it isn't present in their PATH.
We need to add this field to any kubeconfig that gets converted since it can be transported to another machine/environment that does not have kubelogin installed.

### CODE REVIEW

File: pkg/converter/convert.go [hunk 1]
```diff
@@ -199,7 +200,8 @@ func Convert(o Options, pathOptions *clientcmd.PathOptions) error {
 			Args: []string{
 				getTokenCommand,
 			},
-			APIVersion: execAPIVersion,
+			APIVersion:  execAPIVersion,
+			InstallHint: execInstallHint,
```
Comment (weinong) [inline pkg/converter/convert.go:204]: let's use the install hint from input, when provided, before falling back to the hint here

Comment (cirvine-MSFT) [reply pkg/converter/convert.go:205]: Thanks for the suggestion @weinong. I think my latest push contains the changes you wanted. PTAL.

File: pkg/converter/convert.go [hunk 2]
```diff
@@ -51,10 +51,12 @@ const (
 	flagAuthorityHost      = "authority-host"
 	flagFederatedTokenFile = "federated-token-file"
 	flagTokenCacheDir      = "token-cache-dir"
+	flagExecInstallHint    = "install-hint"
 
 	execName        = "kubelogin"
 	getTokenCommand = "get-token"
 	execAPIVersion  = "client.authentication.k8s.io/v1beta1"
+	execInstallHint = "Install kubelogin for use with kubectl by running az aks install-cli"
```
Comment (weinong) [inline pkg/converter/convert.go:59]:
```suggestion
	execInstallHint = "Install kubelogin to connect to Azure AD enabled Kubernetes clusters. Go to https://azure.github.io/kubelogin for details"
```

Comment (weinong) [reply pkg/converter/convert.go:59]: i tend not to put aks context here as this tool can be used for any Azure AD k8s, even on AWS. :)

File: pkg/converter/options.go [hunk 3]
```diff
@@ -33,6 +34,7 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	}
 	fs.StringVar(&o.context, flagContext, "", "The name of the kubeconfig context to use")
 	fs.StringVar(&o.azureConfigDir, flagAzureConfigDir, "", "Azure CLI config path")
+	fs.StringVar(&o.execInstallHint, flagExecInstallHint, "", "The install hint for the kubelogin command when converting to exec format.")
```
Comment (weinong) [inline pkg/converter/options.go:37]: actually, I didn't mean to add an option for the hint. What i meant is, when the kubeconfig to be converted is already of exec format and has install hint, we should preserve it during conversion. For instance
```
az aks get-credentials... 
# the downloaded kubeconfig is in exec format and has aks specific install hint
kubelogin convert-kubeconfig -l msi
# the output should keep the same hint as before
```

File: pkg/converter/convert.go (hunk 2 above)
Comment (cirvine-MSFT) [reply pkg/converter/convert.go]: Makes sense and thanks for that feedback. I did not commit your suggestion as is, but instead aligned with the install hint we are adding to AKS. Let me know if you didn't want that.

File: pkg/converter/options.go (hunk 3 above)
Comment (cirvine-MSFT) [reply pkg/converter/options.go]: Makes sense. Reverted that change.

File: pkg/converter/convert.go [hunk 4]
```diff
@@ -55,6 +55,11 @@ const (
 	execName        = "kubelogin"
 	getTokenCommand = "get-token"
 	execAPIVersion  = "client.authentication.k8s.io/v1beta1"
+	execInstallHint = `
+kubelogin is not installed which is required to connect to AAD enabled cluster.
+
+To learn more, please go to https://aka.ms/aks/kubelogin
```
Comment (weinong) [inline pkg/converter/convert.go:61]: let's use https://azure.github.io/kubelogin/

Comment (weinong) [inline pkg/converter/convert.go:61]:
```suggestion
To learn more, please go to https://azure.github.io/kubelogin/
```

**Review by weinong — APPROVED** [review]
LGTM

**Pull request merged**